rewrite welcome page (fixed histo)

Skylsmoi opened the pull request.

Skylsmoi commented on src/server/kiwix-serve.cpp:
```diff
@@ -879,26 +879,46 @@ int main(int argc, char **argv) {
+"</style>";
+
+  string welcomeBooksHtml = ""
+"<div class='book__list'>";
```
I did it to keep some kind of an indentation for the html.
Creating template in a backend language can quickly make code complicated. I do this to keep the html as much readable as I can.
The other bright side is that the html generated when viewing the source in the browser (right click > inspect source) has a right indentation. A minor addition though.

Skylsmoi commented on src/server/kiwix-serve.cpp:
```diff
@@ -879,26 +879,46 @@ int main(int argc, char **argv) {
+".book__title { font-size: 18px; color: #0645ad; }"
+".book__description { padding: 5px 50px 5px 0px; font-size: 15px; }"
+".book__info { line-height: 18px; color: #777; font-weight: bold; font-size: 13px; }"
+"</style>";
```
Just to keep the html and it's style at the same place, it's easier to code and debugg.
It's not a bad idea to put it in the template file though

kelson42 replied:
Yes, please put as much things as possible in the template.

kelson42 commented:
![screen shot 2017-06-07 at 20 08 50](https://user-images.githubusercontent.com/1029718/26894045-8afefa3e-4bbd-11e7-95b9-ae991f73fa7e.png)
![screen shot 2017-06-07 at 20 08 40](https://user-images.githubusercontent.com/1029718/26894046-8b068cd6-4bbd-11e7-865c-abbeb02f2678.png)

Few remarks:
* The boxes do not have the same height, this should be fix probably by ussing "ellipsis".
* Using ellipsis also to avoid the text collision with the icon
* It would be good to have a nice rendering with 360×640, reducing a bit the marge/padding should be good enough
* An idea, I tend to think that having the page background and boxes greyed/darked and bright if you are over the box would make a better rendering. Currently it's exactly the opposite.
* Having something (light grey?) in background using css3 pattern (http://lea.verou.me/css3patterns/) would really help to make the whole stuff "warmer".

Otherwise looks good to me.

Skylsmoi commented:
In the last commit, I have moved the <style> tag into the template, reduced margin/padding of boxes, implemented the ellipsis css for title and description (so now every boxes will have the same size) and added html "title" on boxes title and description to display the full title or description since there is now an ellipsis

kelson42 commented:
@Skylsmoi Looks good to me! Congrats for all the good work!
